x/ccv/provider/keeper/double_vote.go

```diff
@@ -33,8 +34,18 @@ func (k Keeper) HandleConsumerDoubleVoting(
 		types.NewConsumerConsAddress(sdk.ConsAddress(evidence.VoteA.ValidatorAddress.Bytes())),
 	)
 
-	// execute the jailing
-	k.JailValidator(ctx, providerAddr)
+	k.SlashValidator(ctx, providerAddr)
+	k.JailAndTombstoneValidator(ctx, providerAddr)
+
+	// FIXME: verify the following values
```

Comment:
We currently do not store double voting or misbehaviour evidence. It would be nice to store the evidence when we receive it. However, it's not yet clear how to go about it because `func (EvidenceKeeper) SetEvidence(ctx types.Context, evidence exported.Evidence)` expected evidence is of type `exported.Evidence`. We cannot use `DuplicateVodeEvidence` or `LightClientAttackEvidence` from [evidence.proto](https://github.com/cosmos/cosmos-sdk/blob/5eaa7b8d3c50eefc5dd95210c38c84c1e829509f/proto/tendermint/types/evidence.proto#L4) because they do not implement the `exported.Evidence` interface, e.g., 
```
Cannot use 'tmtypes.LightClientAttackEvidence{}' (type tmtypes.LightClientAttackEvidence) as the type exported.Evidence Type does not implement 'exported.Evidence' as some methods are missing: Reset() ProtoMessage() Route() string
```
We could do something similar to what is shown down below but it doesn't seem so useful. Even today, when querying:
```
gaiad query evidence
evidence:
- '@type': /cosmos.evidence.v1beta1.Equivocation
  consensus_address: cosmosvalcons18eg64zpqxmt6wru5705xg94hvqdh3hvv94e3cm
  height: "12842605"
  power: "225822"
  time: "2022-11-14T12:26:53.319800141Z"
- '@type': /cosmos.evidence.v1beta1.Equivocation
  consensus_address: cosmosvalcons16tnak7apushwznnd3wtku8gm0rt3xytz6ut006
  height: "16172865"
  power: "76511"
  time: "2023-07-17T11:48:22.657230386Z"
pagination:
  next_key: null
  total: "0"
```
the results do not seem that helpful because we do not know what votes led to the equivocation, etc.

Would it make sense to extend the `DuplicateVodeEvidence` or `LightClientAttackEvidence` so they implement the `exported.Evidence` interface?

Reply:
My assumption is that if we do not store evidence, then it would be difficult to explain to a validator for which infraction they got slashes. We would also be consistent with what Cosmos SDK does today [when handling an equivocation](https://github.com/cosmos/cosmos-sdk/blob/5eaa7b8d3c50eefc5dd95210c38c84c1e829509f/x/evidence/keeper/infraction.go#L150).

Reply:
The evidence can be query on the consumer chain as it will still be part of the consumer chain blocks and pass to the evidence module on the consumer.

Reply:
Evidence would be retrieved from the _application state_ of the consumer chain, right? Assuming the application of the consumer chain is not trustworthy, it could have removed the evidence, etc.
Let me know if there are any concerns with the above scenario. For now, I'm removing the storing of evidence on the provider chain.